vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-integration-tests/src/test/java/com/vaadin/flow/component/spreadsheet/test/SheetFilterTableIT.java

```diff
@@ -50,6 +50,22 @@ public void sheetWithFilterTable_rowIsRemoved_filterOptionsAvailable() {
         assertSelectAll(cell);
     }
 
+    @Test
```

Comment:
Since there's only server-side logic involved, an IT feels like an overkill for the purpose.
The following unit test does the job and takes a fraction of the time to run:

in `FilterTableTest.java`
```java
    @Test
    public void filteredTable_unhideRowAndOpenPopup_rowIsUnhidden() {
        getFilterCheckboxGroup().deselect("4");

        spreadsheet.setRowHidden(3, false);
        getPopupButton().openPopup();

        Assert.assertFalse(spreadsheet.isRowHidden(3));
    }
```

Reply:
Good catch! Refactored.